### PR TITLE
SSCS-10766 - Create Set Aside Granted Document  - CALL

### PR DESCRIPTION
**Pull request by DGordonV1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10766
### Change description ###

- Update post hearing handlers to submit grant set aside application review. 
- Some additional util refactoring included. Test classes use placeholder/set_aside types as they're unimportant for the tests being done.
- Passing in caseData to issueDocumentHandler instead of callback (as we were just getting the caseData from the callback anyways), allowed for some refactoring, also added some SscsUtils changes to incorporate some get events/actions/types relating to postHearings. 
- Added both GRANTED & REFUSED for doc templates even though we only have one as they could be separated in future.
- Added refuse related code even though ticket doesn't cover this implementation, it should in theory work the same (not fully tested as out of scope).

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by amoslv1 — APPROVED** [review]

**Review by hajam09 — DISMISSED** [review]

**Review by amoslv1 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/IssueDocumentHandler.java [hunk 1]
```diff
@@ -146,8 +145,17 @@ protected PreSubmitCallbackResponse<SscsCaseData> issueDocument(Callback<SscsCas
 
         final String generatedFileUrl = generateFile.assemble(params);
 
-        documentTypeLabel = documentTypeLabel + ((DRAFT_DECISION_NOTICE.equals(documentType) || DRAFT_ADJOURNMENT_NOTICE.equals(documentType)) ? " generated" : " issued");
-
+        if (DocumentType.SET_ASIDE_APPLICATION.equals(documentType)) {
+            PostHearing postHearing = caseData.getPostHearing();
+            SetAsideActions action = postHearing.getSetAside().getAction();
+            if (GRANT.getValue().equals(action.getCcdDefinition())) {
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/IssueDocumentHandler.java:151]: might be better to change this into a switch statement, will become more extendable for when we come to other post hearing journeys

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/posthearingreview/PostHearingReviewAboutToSubmitHandler.java [hunk 2]
```diff
@@ -57,4 +74,34 @@ protected void updateCaseStatus(SscsCaseData caseData) {
...
+        List<SscsDocument> documents = new ArrayList<>();
+        documents.add(document);
+
+        SscsUtil.addDocumentToBundle(footerService, caseData, documents.get(0), docFileName);
+
+        EventType eventType = getEventTypeFromDocumentReviewTypeAndAction(reviewType, actionName);
+        String templateId = documentConfiguration.getDocuments()
+            .get(caseData.getLanguagePreference())
+            .get(eventType);
+        issueDocument(caseData, documentType, templateId, generateFile, userAuthorisation);
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/posthearingreview/PostHearingReviewAboutToSubmitHandler.java:105]: does this actually issue the document? To me it does seem like it makes the document but then adds it as the preview document which field we clear on line 63

File: src/main/java/uk/gov/hmcts/reform/sscs/util/SscsUtil.java [hunk 3]
```diff
@@ -64,10 +61,65 @@ public static void addDocumentToDocumentTab(FooterService footerService, SscsCas
...
+
+    @Nullable
+    public static EventType getEventTypeFromDocumentReviewTypeAndAction(PostHearingReviewType reviewType,
+                                                                           String actionName) {
+        if (isNull(reviewType) || isNull(actionName)) {
+            return null;
+        }
+        // TODO: fill in other event types for other post hearing routes, use switch if desired
+        if (SET_ASIDE.equals(reviewType)) {
+            if (GRANT.getValue().equals(actionName)) {
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/util/SscsUtil.java:117]: switch statement

**Review by nickhill111 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/IssueDocumentHandler.java (hunk 1 above)
Comment (DGordonV1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/IssueDocumentHandler.java]: Done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/posthearingreview/PostHearingReviewAboutToSubmitHandler.java (hunk 2 above)
Comment (DGordonV1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/posthearingreview/PostHearingReviewAboutToSubmitHandler.java:103]: Done. Doc won't be getting cleared anymore.

**Review by jarekPierchala — DISMISSED** [review]

**Review by AndreiTrushkov — APPROVED** [review]

**Review by amoslv1 — APPROVED** [review]

**Pull request closed without merging**